### PR TITLE
Bip70

### PR DESCRIPTION
**Pull request by thehobbit85**
Add Bip70 Support for the plugin.
Also improve some of the testing suite

### CODE REVIEW

File: src/utils/paymentRequest.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// @flow
+import bcoin from 'bcoin'
+import parse from 'url-parse'
+// $FlowFixMe
+import type { AbcPaymentProtocolInfo } from 'edge-core-js'
```
Comment (paullinator) [inline src/utils/paymentRequest.js]: Use "Edge" versions

File: src/utils/paymentRequest.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { toNewFormat } from './addressFormat/addressFormatIndex.js'
+
+// $FlowFixMe
+export const parsePayment = (
+  paymentBuffer: Buffer,
+  network: string,
+  currencyCode: string
+): AbcPaymentProtocolInfo => {
```
Comment (paullinator) [inline src/utils/paymentRequest.js]: Use "Edge" versions

File: src/utils/paymentRequest.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    nativeAmount += jsonObj.value
+    abcSpendTarget.push({
+      currencyCode: currencyCode,
+      publicAddress: toNewFormat(jsonObj.address, network),
+      nativeAmount: `${jsonObj.value}`
+    })
+  }
+
+  const abcPaymentProtocolInfo = {
+    nativeAmount: `${nativeAmount}`,
```
Comment (paullinator) [inline src/utils/paymentRequest.js]: BIP21 specifies amounts in exchangeAmount (ie BTC) not satoshis. Does bcoin for sure return nativeAmounts?

Comment (thehobbit85) [reply src/utils/paymentRequest.js]: I'm pretty sure. I think the test covers that

**Pull request merged**